### PR TITLE
Compose email with text style for mobile

### PR DESCRIPTION
**Pull request by ManhNTX**
https://user-images.githubusercontent.com/107173849/179194295-7e8ba6cf-3df0-4152-801f-16e12e2ad4e2.mov

https://user-images.githubusercontent.com/107173849/179436303-83ecafa8-c1cf-4d3f-b570-ac3bdde95c68.mov

### CODE REVIEW

File: lib/features/composer/presentation/composer_controller.dart [hunk 1]
```diff
@@ -110,6 +112,22 @@ class ComposerController extends BaseController {
...
+  selectTextStyleType(RichTextStyleType richTextStyleType){
+    if(listTextStyleApply.contains(richTextStyleType)){
+      listTextStyleApply.remove(richTextStyleType);
+    } else {
+      listTextStyleApply.add(richTextStyleType);
+    }
+  }
+
+  bool isTextStyleTypeSelected(RichTextStyleType richTextStyleType){
+    if(listTextStyleApply.contains(richTextStyleType)){
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:124]: You can abbreviate it like this: `return listTextStyleApply.contains(richTextStyleType)`

**Review by dab246 — APPROVED** [review]

File: lib/features/composer/presentation/model/rich_text_style_type.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import 'package:core/core.dart';
+import 'package:html_editor_enhanced/html_editor.dart';
+
+enum RichTextStyleType {
+  bold,
+  italic,
+  underline;
+
+  String get commandAction {
```
Comment (hoangdat) [inline lib/features/composer/presentation/model/rich_text_style_type.dart:9]: I did not see this getter use anywhere :)

**Comment by dab246** [comment]
Screen Shot 2022-07-18 at 13 51 24

You pay attention to update the color of the selected icon to be the same as the design.

File: lib/features/composer/presentation/composer_view.dart [hunk 3]
```diff
@@ -490,6 +496,57 @@ class ComposerView extends GetWidget<ComposerController> with AppLoaderMixin {
...
+
+  Widget _buildIconStyleText({
+    required String path,
+    required bool? isSelected,
+    required VoidCallback onTap,
+  }){
+    return buildIconWeb(
+        icon: SvgPicture.asset(
+            path,
+            color: isSelected == true ? AppColor.colorNameEmail : AppColor.colorDividerMailbox,
```
Comment (hoangdat) [inline lib/features/composer/presentation/composer_view.dart:541]: Should not do that, please define the dedicated color for your widgets. What happens in the future if you change the color of `NameEmail`? You will be hard to manage the color of other widgets like that.

In this case, enable/disable color for style button can be the same.

File: core/lib/presentation/extensions/color_extension.dart [hunk 4]
```diff
@@ -69,6 +69,7 @@ extension AppColor on Color {
   static const bgWordSearch = Color(0x3D007AFF);
   static const lineItemListColor = Color(0xFF99A2AD);
   static const colorNameEmail = Color(0xFF000000);
+  static const colorRichTextStyleSelected = Color(0xFF000000);
```
Comment (hoangdat) [inline core/lib/presentation/extensions/color_extension.dart:72]: I did not see where you use it

**Pull request merged**